bitswap/provider.go

```diff
@@ -26,36 +26,63 @@ import (
 	"go.opencensus.io/tag"
 )
 
-var logger = log.Logger("autoretrieve")
+var log = logging.Logger("provider")
 
 // Wantlist want type redeclarations
 const (
 	wantTypeHave  = bitswap_message_pb.Message_Wantlist_Have
 	wantTypeBlock = bitswap_message_pb.Message_Wantlist_Block
 )
 
-// Task queue topics
-type (
-	topicHave     cid.Cid
-	topicDontHave cid.Cid
-	topicBlock    cid.Cid
+// Response queue actions
+type ResponseAction uint
+
+type ResponseData struct {
+	action ResponseAction
+	reason string
+}
+
+const (
+	sendHave ResponseAction = iota
+	sendDontHave
+	sendBlock
```

Comment:
when encountering these values down below I found it a bit confusing what they were; can I suggest we extend their names to explain what they are? `sendHaveAction`, `sendDontHaveAction` or something like that?